gremlin-go/driver/graphBinary.go

```diff
@@ -583,6 +578,29 @@ func pathReader(buffer *bytes.Buffer, typeSerializer *graphBinaryTypeSerializer)
 	return p, nil
 }
 
+// Format: Same as List.
+// Mostly similar to listWriter and listReader with small changes
+func setWriter(value interface{}, buffer *bytes.Buffer, typeSerializer *graphBinaryTypeSerializer) ([]byte, error) {
+	slice, err := value.(SetTemplate).convertSetToSlice()
+	sliceFiltered := removeDuplicateValues(slice)
```

Comment:
newbie question: why error check (line 586) placed after using result (line 585)?

Reply:
Nice catch! thats an error on my part. Should check for error before using the value.